src/openapi/streaming/connection/connection.ts

```diff
@@ -0,0 +1,361 @@
+            this.unauthorizedCallback = this.ensureValidState.bind(
+                this,
+                callback,
+                'unauthorizedCallback',
+            );
+            this.transport.setUnauthorizedCallback(this.unauthorizedCallback);
+        }
+    }
+
+    setStateChangedCallback(callback: StateChangeCallback) {
+        if (this.transport) {
+            this.stateChangedCallback = this.ensureValidState.bind(
+                this,
+                callback,
+                'stateChangedCallback',
+            );
+            this.transport.setStateChangedCallback(this.stateChangedCallback);
+        }
+    }
+
+    setReceivedCallback(callback: ReceiveCallback) {
+        if (this.transport) {
+            this.receiveCallback = this.ensureValidState.bind(
+                this,
+                callback,
+                'receivedCallback',
+            );
+            this.transport.setReceivedCallback(this.receiveCallback);
+        }
+    }
+
+    setConnectionSlowCallback(callback: Callback) {
+        if (this.transport) {
+            this.connectionSlowCallback = this.ensureValidState.bind(
+                this,
+                callback,
+                'connectionSlowCallback',
+            );
+            this.transport.setConnectionSlowCallback(
+                this.connectionSlowCallback,
+            );
+        }
+    }
+
+    start(callback: Callback) {
+        if (this.transport) {
+            this.state = STATE_STARTED;
+            this.startCallback = callback;
+
+            const transportOptions = {
+                ...this.transports[this.transportIndex as number].options,
+                ...this.options,
+            };
+            this.transport.start(transportOptions, this.startCallback);
+        }
+    }
+
+    stop() {
+        if (this.transport) {
+            this.state = STATE_STOPPED;
+            this.transport.stop();
+        }
+    }
+
+    dispose() {
+        this.state = STATE_DISPOSED;
+    }
+
+    updateQuery(
+        authToken: string,
+        contextId: string,
+        authExpiry?: number,
+        forceAuth = false,
+    ) {
+        this.authToken = authToken;
+        this.contextId = contextId;
+        this.authExpiry = authExpiry;
+
+        log.debug(LOG_AREA, 'Connection update query', {
+            contextId,
+            authExpiry,
+        });
+
+        if (this.transport) {
+            this.transport.updateQuery(
+                this.authToken,
+                this.contextId,
+                this.authExpiry,
+                forceAuth,
+            );
+        }
+    }
+
+    getQuery() {
+        if (this.transport) {
+            return this.transport.getQuery?.();
+        }
+    }
+
+    onOrphanFound() {
+        if (this.transport?.onOrphanFound) {
+            this.transport.onOrphanFound();
+        }
+    }
+
+    onSubscribeNetworkError() {
+        if (this.transport && this.transport.onSubscribeNetworkError) {

```

Comment:
nitpick - but 3 different patterns here for the exact same thing - call a function on a object where object or function may be undefined. Would be nice to be consistent.

Reply:
Fixed